File: Modules/Kit/Common/UrgentOrder/SIE.Web.Kit.UrgentOrder/ItemUrgentOrders/Commands/ItemUrgentOrderAddCommand.js
SIE.defineCommand('SIE.Web.Kit.UrgentOrder.ItemUrgentOrders.Commands.ItemUrgentOrderAddCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "iconfont icon-AddEntity icon-green" },
    onItemCreated: function (entity) {
        if (entity) {
            var model = entity.data;
            var me = this;
            this.view.execute({
                data: model,
                success: function (res) {
                    var data = res.Result;
                    entity.setNo(data.No);
                    entity.setOrderState(data.OrderState);
                    entity.setQty(data.Qty);
                    entity.setDemandTime(data.DemandTime);
                }
            }, me.view);
        }
        this.mon(entity, 'propertyChanged', this._onEntityPropertyChanged, this);
    },
    _onEntityPropertyChanged: function (e) {
        var me = this;
        var data = e.entity.data;
        if (e.property.length > 0) {
            if (e.property.indexOf('IsReceive') >= 0) {
                if (entity.getIsReceive()) {
                    entity.setReceiveQty(entity.getQty());
                }
            }
        }
    }
})

File: Modules/Kit/Common/UrgentOrder/SIE.Web.Kit.UrgentOrder/ItemUrgentOrders/Commands/ItemUrgentOrderAddCommand.cs
using SIE.Kit.UrgentOrder.ItemUrgentOrders;
using SIE.Web.Command;
using System;

namespace SIE.Web.Kit.UrgentOrder.ItemUrgentOrders.Commands
{
    /// <summary>
    /// 物料加急单添加命令
    /// </summary>
    public class ItemUrgentOrderAddCommand : ViewCommand
    {
        protected override object Excute(ViewArgs args, string scope)
        {
            var data = args.Data.ToJsonObject<ItemUrgentOrder>();
            data.No = RT.Service.Resolve<ItemUrgentOrderController>().GetItemUrgentOrderNo();
            data.OrderState = UrgentOrderState.Valid;
            data.Qty = 1000;
            data.DemandTime = DateTime.Now;
            data.CreateDate = DateTime.Now;
            return data;
        }
    }
}
